pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -94,8 +93,10 @@ func (c *Reconciler) cleanupAffinityAssistants(pr *v1beta1.PipelineRun) error {
 	return errorutils.NewAggregate(errs)
 }
 
-func getAffinityAssistantName(pipelineWorkspaceName string, owner metav1.OwnerReference) string {
-	return fmt.Sprintf("%s-%s", pipelineWorkspaceName, owner.Name)
+func getAffinityAssistantName(pipelineWorkspaceName string, pipelineRunName string) string {
+	hashBytes := sha256.Sum256([]byte(pipelineWorkspaceName + pipelineRunName))
+	hashString := fmt.Sprintf("%x", hashBytes)
+	return fmt.Sprintf("%s-%s", "affinity-assistant", hashString[:10])
```

Comment:
we actually have some existing libs around this that you might be able to use: https://github.com/tektoncd/pipeline/blob/master/pkg/names/generate.go

Reply:
I think I had a look at it, but it is "random"

>random suffix

I my use case, I want a consistent hash - from the Workspace Name + PipelineRun Name - so that the same combination always get the same hash.